Add support of alternative PHP configuration instead of composer.json (#104)
yiisoft/config: README.md

```json
"extra": {
    "config-plugin-options": {
      "source-directory": "config"
    },
    "config-plugin": {
        // ...
    }
},
```

The `source-directory` option specifies where to read the configs from for a package the option is specified for.
It is available for all packages, including the root package, which is typically an application.

> The value is a path relative to where the `composer.json` file is located. The default value is empty string.

Reviewer: ```suggestion
The value is a path relative to where the `composer.json` file is located. The default value is an empty string.
```

If defined in an environment, `params` will be merged with `params` from the main configuration,
and could be used as `$params` in all configurations.

## Configuration in a PHP file

> You can define the configuration in a PHP file, in the `extra` section of the `composer.json`,
> specify only the path to the PHP file:

Reviewer: ```suggestion
You can define configuration in a PHP file. To do it, specify a PHP file path in the `extra` section of the `composer.json`:
```

```json
"extra": {
    "config-plugin-file": "path/to/configuration/file.php"
},
```

> Configurations are specified in the same way only in PHP format:

Reviewer: ```suggestion
Configurations are specified in the same way, only in PHP format:
```

```php
return [
    'config-plugin-options' => [
        'source-directory' => 'config',  
    ],
    'config-plugin' => [
        'params' => [
            'params.php',
            '?params-local.php',
        ],
        'web' => 'web.php', 
    ],
    'config-plugin-environments' => [
        'dev' => [
            'params' => 'dev/params.php',
            'app' => [
                '$web',
                'dev/app.php',
            ],
        ],
        'prod' => [
            'app' => 'prod/app.php',
        ],
    ],
];
```

If you specify the file path, the remaining sections (`config-plugin-*`) in `composer.json` will be ignored
and configurations will be read from the specified PHP file. The path to the PHP configuration

> file is considered relative to where the composer.json `file is located.`

Reviewer 2: ```suggestion
file is considered relative to where the `composer.json` file is located.
```

> If you specify the file path, the remaining sections (`config-plugin-*`) in `composer.json` will be ignored
> and configurations will be read from the specified PHP file. The path to the PHP configuration
> file is considered relative to where the `composer.json` file is located.

Reviewer: ```suggestion
If you specify the file path, the remaining sections (`config-plugin-*`) in `composer.json` will be ignored
and configurations will be read from the PHP file specified. The path is relative to where the `composer.json` file is located.
```
